### PR TITLE
[skip-changelog] Add query parameters to urls generated by lib commands

### PR DESCRIPTION
**Pull request by MatteoPologruto**
## Please check if the PR fulfills these requirements

See [how to contribute](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/)

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [x] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] `UPGRADING.md` has been updated with a migration guide (for breaking changes)

## What kind of change does this PR introduce?
Code enhancement

## What is the new behavior?
Three different query parameters have been added to the urls generated by `lib` commands like `lib install` and `lib upgrade`. The query parameters are:
- `install`: which is selected when a library is installed through `lib install`;
- `upgrade`: which is selected when a library is upgraded through `lib upgrade`;
- `depends`: which is selected when a library dependency is installed.

## Does this PR introduce a breaking change, and is [titled accordingly](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#breaking)?
No

### CODE REVIEW

File: arduino/httpclient/httpclient.go [hunk 1]
```diff
@@ -25,14 +25,19 @@ import (
...
 	rpc "github.com/arduino/arduino-cli/rpc/cc/arduino/cli/commands/v1"
 	"github.com/arduino/go-paths-helper"
+	"github.com/sirupsen/logrus"
 	"go.bug.st/downloader/v2"
 )
 
 var tr = i18n.Tr
 
 // DownloadFile downloads a file from a URL into the specified path. An optional config and options may be passed (or nil to use the defaults).
 // A DownloadProgressCB callback function must be passed to monitor download progress.
```
Comment (umbynos) [inline arduino/httpclient/httpclient.go:35]: Could you please change this comment and include info regarding the new parameter?

File: arduino/resources/download.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ import (
 
 // Download performs a download loop using the provided downloader.Config.
 // Messages are passed back to the DownloadProgressCB using label as text for the File field.
```
Comment (umbynos) [inline arduino/resources/download.go:29]: same as above

File: commands/lib/download.go [hunk 3]
```diff
@@ -31,7 +31,7 @@ import (
 var tr = i18n.Tr
 
 // LibraryDownload FIXMEDOC
```
Comment (umbynos) [inline commands/lib/download.go:33]: This could be the chance to fix this

File: commands/lib/install.go [hunk 4]
```diff
@@ -31,7 +31,7 @@ import (
 )
 
 // LibraryInstall FIXMEDOC
```
Comment (umbynos) [inline commands/lib/install.go:33]: same as above

File: commands/lib/install.go [hunk 5]
```diff
@@ -96,11 +96,20 @@ func LibraryInstall(ctx context.Context, req *rpc.LibraryInstallRequest, downloa
 	}
 
 	for libRelease, installTask := range libReleasesToInstall {
-		if err := downloadLibrary(lm, libRelease, downloadCB, taskCB); err != nil {
-			return err
-		}
-		if err := installLibrary(lm, libRelease, installTask, taskCB); err != nil {
-			return err
+		if libRelease.GetName() == req.Name {
```
Comment (umbynos) [inline commands/lib/install.go:99]: why is this requires? It's not so clear

File: internal/cli/lib/download.go [hunk 6]
```diff
@@ -60,7 +60,7 @@ func runDownloadCommand(cmd *cobra.Command, args []string) {
 			Name:     library.Name,
 			Version:  library.Version,
 		}
-		_, err := lib.LibraryDownload(context.Background(), libraryDownloadRequest, feedback.ProgressBar())
+		_, err := lib.LibraryDownload(context.Background(), libraryDownloadRequest, feedback.ProgressBar(), "")
```
Comment (umbynos) [inline internal/cli/lib/download.go:63]: I don't know if this has to be left empty

**Review by umbynos — CHANGES_REQUESTED** [review]
left some suggestions

**Review by umbynos — APPROVED** [review]
LGTM :checkered_flag:

**Comment by cmaglie** [comment]
I see a couple of problems with this PR:
* it changes the signature of the function `commands.LibraryDownload`, this is a breaking change that is not documented
* all public functions in package `commands` are made to strictly follow the gRPC API. If we need to add a parameter in any of the public functions of `command.*` we must first change the gRPC API and rebuild it otherwise, we allow the CLI to bypass the gRPC API. In fact, this implementation works only in the CLI, there is no way to access this feature from the gRPC API (so it will not work in the Arduino IDE or any other gRPC client).

**Pull request merged**